master/buildbot/steps/source/base.py

```diff
@@ -279,22 +280,20 @@ def start(self):
                 # root is optional.
                 patch = s.patch
                 if patch:
-                    self.addCompleteLog("patch", bytes2unicode(patch[1]))
+                    yield self.addCompleteLog("patch", bytes2unicode(patch[1]))
             else:
                 log.msg("No sourcestamp found in build for codebase '{}'".format(self.codebase))
-                self.step_status.setText(
-                    ["Codebase", '{}'.format(self.codebase), "not", "in", "build"])
-                self.addCompleteLog("log",
-                                    "No sourcestamp found in build for codebase '{}'".format(
-                                            self.codebase))
+                self.descriptionDone = ["Codebase", '{}'.format(self.codebase),
```

Comment:
the '{}'.format looks suspicious. probably str(self.codebase) is better..

Reply:
Fixed in e4d76550e1.